test/unittest/rtps/builtin/BuiltinDataSerializationTests.cpp

```diff
@@ -2186,6 +2186,181 @@ TEST(BuiltinDataSerializationTests, interoperability_with_intercomdds)
     }
 }
 
+/*!
+ * This test checks that Fast DDS can properly serialize ResourceLimitsQos in DATA(w) when it has more fields than
+ * max_samples, max_instances and max_samples_per_instance. It also checks that those fields are not serialized
+ * into the allocated_samples and extra samples fields.
+ */
+TEST(BuiltinDataSerializationTests, interoperability_with_intercomdds_writer_resource_limits)
+{
+    const VendorId_t intercom_vendor_id = { 1, 5 };
```

Comment:
We don't need to use intercomdds, since we are inventing the resource_limits part here.

I would use {0, 0} as the vendor id, and rename things.

```suggestion
TEST(BuiltinDataSerializationTests, interoperability_with_other_vendor_writer_resource_limits)
{
    const VendorId_t other_vendor_id = { 0, 0 };
```